front/src/modules/trainschedule/components/DriverTrainScheduleV2/driverTrainScheduleV2ExportCSV.ts

```diff
@@ -0,0 +1,245 @@
+  electrificationProfile?: string;
+};
+
+const pointToComma = (number: number) => number.toString().replace('.', ',');
+const compareOldActualValues = (old?: string, actual?: string) =>
+  old !== undefined && actual === undefined ? old : actual;
+
+const getStepSpeedLimit = (_position: number, speedLimitList: Train['vmax']) => {
+  const bisector = d3.bisectLeft(
+    speedLimitList.map((d: SpeedPosition) => d.position),
+    _position
+  );
+  return speedLimitList[bisector].speed || 0;
+};
+
+// Add OPs inside speedsteps array, gather speedlimit with stop position, add electrification ranges,
+// and sort the array along position before return
+const overloadSteps = (
+  trainRegime: ReportTrainV2,
+  operationalPoints: OperationalPointWithTimeAndSpeed[],
+  speedLimits: SpeedPosition[],
+  electrificationRanges: ElectrificationRangeV2[]
+): PositionSpeedTimeOP[] => {
+  const speedsAtOps = operationalPoints.map((op) => ({
+    position: op.position,
+    speed: op.speed,
+    time: op.time,
+    op: op.name,
+    ch: op.ch,
+    lineCode: op.line_code,
+    trackName: op.track_name,
+  }));
+  const speedsAtSpeedLimitChange = speedLimits.map((speedLimit) => ({
+    position: mmToM(speedLimit.position),
+    speed: interpolateValue(trainRegime, speedLimit.position, 'speeds'),
+    speedLimit: speedLimit.speed,
+    time: interpolateValue(trainRegime, speedLimit.position, 'times'),
+  }));
+  const speedsAtElectrificationRanges: PositionSpeedTimeOP[] = [];
+  electrificationRanges.forEach((electrification, idx) => {
+    const electrificationType = electrification.electrificationUsage.type;
+    const electricalProfileType = electrification.electrificationUsage.electrical_profile_type;
+
+    const electrificationMode =
+      electrificationType === 'electrification' ? electrification.electrificationUsage.voltage : '';
+    const electrificationProfile =
+      electrificationType === 'electrification' &&
+      electricalProfileType === 'profile' &&
+      electrification.electrificationUsage.profile
+        ? electrification.electrificationUsage.profile
+        : '';
+    const electrificationStart = electrification.start;
+
+    speedsAtElectrificationRanges.push({
+      position: mmToM(electrificationStart),
+      speed: interpolateValue(trainRegime, electrificationStart, 'speeds'),
+      electrificationType,
+      electrificationMode,
+      electrificationProfile,
+      time: interpolateValue(trainRegime, electrificationStart, 'times'),
+    });
+
+    // ElectrificationRanges could not be continuous, so we've to handle the case of
+    // empty ranges: when previousStop isn't equal to nextStart, we add a one-meter away
+    // step with empty values to ensure correct spread of information along all steps
+    if (
+      electrificationRanges[idx + 1] &&
+      electrification.stop < electrificationRanges[idx + 1].start
+    ) {
+      speedsAtElectrificationRanges.push({
+        position: mmToM(electrification.stop + 1),
```

Comment:
Why do we need to add 1 meter here ?
```suggestion
        position: mmToM(electrification.stop + 1000),
```

Reply:
Not sure why it was needed. But the electrification positions are already in meters when converted in useSimulationResults. I will put a comment in PathPropertiesFormatted type.

Reply:
You're right